src/main/java/org/junitpioneer/jupiter/ReportEntry.java

```diff
@@ -45,4 +45,36 @@
 	 */
 	String value();
 
+	/**
+	 * Specifies when the extension should publish the report entry.
+	 * Defaults to ReportEntry.Timing.ALWAYS
+	 * @see PublishCondition
+	 */
+	PublishCondition when() default PublishCondition.ALWAYS;
+
+	/**
+	 * The available parameters that define when the extension
+	 * should publish the report entry.
+	 * <p>
+	 * Available parameters are:
+	 *  - ALWAYS:		publish report entry before method call
+	 *  - ON_SUCCESS:	publish report entry after successful method run
+	 *  - ON_FAILURE:	publish report entry after failed test run
+	 *  - NEVERMORE:	never publishes report entry
+	 */
+	enum PublishCondition {
+		ALWAYS, ON_SUCCESS, ON_FAILURE, NEVERMORE
+	}
+
+	/**
+	 * This makes the {@code ReportEntry} repeatable.
+	 */
+	@Retention(RetentionPolicy.RUNTIME)
```

Comment:
Note: I appreciate the declartion of the repeatable annotation inside it (as I suggested it in #215 ). But as the discussion on that has not really started I can not say if this is fine or has to be changed later.

Reply:
I see no reason to store this is a separate file. You can not use it by itself, only in combination with the `@ReportEntry`.